### PR TITLE
refactor(core): remove unused logic from reflection capabilities

### PR DESCRIPTION
**Pull request by JoostK**
This commit removes a bunch of methodss from `ReflectionCapabilities` as they
have gone unused. This also removes `Reflector` as it doesn't serve any purpose
and it is not exposed as public API, so can be safely removed.

### CODE REVIEW

**Comment by Nutza974** [comment]
Reply

ในวันที่ อา. 13 มี.ค. 2022 04:21 น. Joost Koehoorn ***@***.***>
เขียนว่า:

File: modules/benchmarks/src/old/di/di_benchmark.ts [hunk 1]
```diff
@@ -7,23 +7,16 @@
  */
 
 import {Injectable, ReflectiveInjector, ReflectiveKey} from '@angular/core';
-import {reflector} from '@angular/core/src/reflection/reflection';
```
Comment (jessicajaniuk) [inline modules/benchmarks/src/old/di/di_benchmark.ts:10]: Without this, how will Angular ride a bike safely at night?

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

File: modules/benchmarks/src/old/di/di_benchmark.ts (hunk 1 above)
Comment (JoostK) [reply modules/benchmarks/src/old/di/di_benchmark.ts:10]: Just a lot of fines incoming, Jess. I hope Angular is good friends with law enforcement.

**Comment by dylhunn** [comment]
This PR has merge conflicts with patch, so I'm landing it on `master` for now. Let me know if you'd like to merge a rebased patch port as well.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 71ee417fae5e46253a33dd9b5b5032903f3042fc.

**Pull request closed without merging**